pingcap/tidb
# execution: commit the transaction before responding explain analyze results to the client

### What problem does this PR solve?


Issue Number: close #37373 

Problem Summary:
The explain analyze statement execution with DML executors may respond to the client before the transaction commit finishing.

### What is changed and how it works?
Trigger the commit operation if the statement is an `explain analyze` statement with DML executors before returning the record set of itself.

### Check List

Tests 

   -  Sleep before the commit
   -  Execute explain analyze
   - Check if it responses before the commit has finished

Side effects



Documentation



### Release note



Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
Fix the issue explain analyze with DML executors may respond to the client before the transaction commit has finished.
```
